### PR TITLE
feat: Remove 'active' field from OAuth credentials

### PR DESCRIPTION
**Pull request by boxbeam**
Removes the active field and determines active status based on whether the credential is present or not

### CODE REVIEW

**Comment by wsxiaoys** [comment]
add @darknight for review

File: ee/tabby-webserver/src/schema/mod.rs [hunk 1]
```diff
@@ -410,6 +409,19 @@ impl Mutation {
             .await?;
         Ok(true)
     }
+
+    async fn delete_email_service_credential(ctx: &Context) -> FieldResult<bool> {
```
Comment (darknight) [inline ee/tabby-webserver/src/schema/mod.rs:413]: remove unrelated method

File: ee/tabby-webserver/src/schema/mod.rs [hunk 2]
```diff
@@ -410,6 +409,19 @@ impl Mutation {
...
+        ctx.locator
+            .email_service_credential()
+            .delete_email_service_credential()
+            .await?;
+        Ok(true)
+    }
+
+    async fn delete_oauth_credential(ctx: &Context, provider: OAuthProvider) -> FieldResult<bool> {
```
Comment (darknight) [inline ee/tabby-webserver/src/schema/mod.rs:421]: Use defined `type Result<T, E = CoreError>` if there's no particular reason.

File: ee/tabby-webserver/src/schema/mod.rs (hunk 1 above)
Comment (wsxiaoys) [reply ee/tabby-webserver/src/schema/mod.rs:413]: it's related (delete means deactive)

File: ee/tabby-db/src/github_oauth_credential.rs [hunk 3]
```diff
@@ -48,15 +45,14 @@ impl DbConn {
...
-                    ":active": active,
                     })?;
                     Ok(())
                 })
                 .await?;
             Ok(())
         } else {
             let sql = r#"
-            UPDATE github_oauth_credential SET client_id = :cid, active = :active, updated_at = datetime('now')
+            UPDATE github_oauth_credential SET client_id = :cid, updated_at = datetime('now')
```
[the remark this answers is not in the document]
Comment (wsxiaoys) [reply ee/tabby-db/src/github_oauth_credential.rs:55]: i think it's still necessary to support Option<&str> - for example, an user might update smtp server url / port / protocol without updating client secret.

File: ee/tabby-webserver/src/schema/mod.rs [hunk 4]
```diff
@@ -363,7 +363,7 @@ impl Mutation {
         provider: OAuthProvider,
         client_id: String,
         client_secret: Option<String>,
-        active: bool,
+        _active: bool,
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/mod.rs:366]: remove this

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: ee/tabby-db/src/github_oauth_credential.rs [hunk 5]
```diff
@@ -79,13 +74,26 @@ impl DbConn {
         }
     }
 
+    pub async fn delete_github_oauth_credential(&self) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (boxbeam) [reply ee/tabby-db/src/github_oauth_credential.rs:77]: Added a test case

File: ee/tabby-db/src/github_oauth_credential.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (darknight) [reply ee/tabby-db/src/github_oauth_credential.rs]: `smtp`? so we want to manage `smtp credential` & `oauth credential` together?

File: ee/tabby-webserver/src/service/auth.rs [hunk 6]
```diff
@@ -403,15 +400,20 @@ impl AuthenticationService for DbConn {
         &self,
         provider: OAuthProvider,
         client_id: String,
-        client_secret: Option<String>,
-        active: bool,
+        client_secret: String,
     ) -> Result<()> {
         match provider {
             OAuthProvider::Github => Ok(self
-                .update_github_oauth_credential(&client_id, client_secret.as_deref(), active)
```
Comment (darknight) [inline ee/tabby-webserver/src/service/auth.rs:411]: overall looks good to me. but you may need to revert this change if we still need to pass `Option<&str>`.

Comment (wsxiaoys) [reply ee/tabby-webserver/src/service/auth.rs:411]: for this specific case (github oauth), i think we don't really have a case that need to update client_id only, without updating the client_secret. thus I think it's fine to make client_secret a required field in updating, after removing active.

Comment (darknight) [reply ee/tabby-webserver/src/service/auth.rs:411]: sounds good, no more comments from my side.
